@here/harp-vectortile-datasource/lib/VectorTileDecoder.ts

```diff
@@ -56,28 +56,24 @@ export class VectorTileDataProcessor implements IGeometryProcessor {
         private readonly m_projection: Projection,
         private readonly m_styleSetEvaluator: StyleSetEvaluator,
         private readonly m_showMissingTechniques: boolean,
+        private readonly m_dataAdapter: DataAdapter,
         private readonly m_dataFilter?: OmvFeatureFilter,
         private readonly m_featureModifiers?: OmvFeatureModifier[],
         private readonly m_gatherFeatureAttributes = false,
         private readonly m_skipShortLabels = true,
         private readonly m_storageLevelOffset = 0,
         private readonly m_enableElevationOverlay = false,
-        private readonly m_roundUpCoordinatesIfNeeded = false,
+        roundUpCoordinatesIfNeeded = false,
         private readonly m_languages?: string[]
     ) {
-        const styleSetDataFilter = new StyleSetDataFilter(m_styleSetEvaluator);
-        const dataPreFilter = m_dataFilter
-            ? new ComposedDataFilter([styleSetDataFilter, m_dataFilter])
-            : styleSetDataFilter;
-
-        // Register the default adapters.
-
-        const omvDataAdapter = new OmvDataAdapter(this, dataPreFilter, logger);
-        omvDataAdapter.roundUpCoordinatesIfNeeded = m_roundUpCoordinatesIfNeeded;
-        this.m_dataAdapters.push(omvDataAdapter);
-
-        this.m_dataAdapters.push(new GeoJsonVtDataAdapter(this, dataPreFilter, logger));
-        this.m_dataAdapters.push(new GeoJsonDataAdapter(this, dataPreFilter, logger));
+        if (this.m_dataAdapter instanceof OmvDataAdapter) {
```

Comment:
Can this be done outside this class?

I find it a bit ugly that VectorTileDataProcessor has to know about the OmvDataAdapter internals.

Reply:
I've moved all OMV-specific decode options to OmvDataAdapter.